libs/shared-web/src/hooks/useAuthController.ts

```diff
@@ -1,13 +1,21 @@
 import { useEffect, useCallback, useState } from 'react';
-import axios from 'axios';
 import jwtDecode from 'jwt-decode';
-import { useNavigate } from 'react-router-dom';
+import { useLocation, useNavigate } from 'react-router-dom';
 import { useQuery, useQueryClient } from '@tanstack/react-query';
 import * as Sentry from '@sentry/react';
 import type { IJwtPayload, IOrganizationEntity, IUserEntity } from '@novu/shared';
 
 import { useSegment } from '../providers';
 import { api } from '../api';
+import { ROUTES } from '../constants';
+
+const LOCAL_STORAGE_AUTH_TOKEN_KEY = 'auth_token';
+const UNAUTHENTICATED_STATUS_CODE = 401;
+
+export interface IUserWithContext extends IUserEntity {
+  organizationId?: string;
+  environmentId?: string;
+}
```

Comment:
This object now contains the current context for the User's active Organization & Environment identifiers.

Reply:
Nit-picky comment: I'd love us to have an AuthContext object that contains:

```
{
  user,
  user_id,
  organization,
  org_id,
  environment,
  ...roles,
  ...
}```

That way, we will be set for an easier auth migration/integration and we will have everything in one place.